### DISCUSSION TITLE
RainTree Python Sim Question

### DISCUSSION THREAD

**Discussion by jeremybeal11** [pokt-network/pocket#345] [Q&A]
When I run the PY sim, it generates the tree with three branches with different nodes( with one of the branches referencing the orientation node). @Olshansk, can you help me understand the output of the sim? 

Steps to reproduce:
run `rainTreeTestOutputFilename=/tmp/answer.go numRainTreeNodes=6 make p2p_test_generator`

```
###################

                                val_1                                                              
           ┌──────────────────────┴────────────┬────────────────────────────────┐                  
         val_3                               val_1                            val_5                
   ┌───────┴────┬─────────┐            ┌───────┴────┬─────────┐         ┌───────┴────┬─────────┐   
 val_4        val_3     val_5        val_2        val_1     val_3     val_6        val_5     val_1 

###################
```

**Comment by Olshansk** [comment]
See slide 13 of this deck and lmk if that's enough info or if you need additional clarifying details: https://github.com/pokt-network/pocket/files/9853354/raintree.pdf

Also, CCing @BenVanGithub (the author of RainTree) to help out with answering questions in case I'm not available.

**Comment by BenVanGithub** [comment]
Short answer… when a node finishes sending it’s two messages, it becomes a third recipient and behaves just like it would if it had received that message from someone else.  Clear like mud?

**Reply by jeremybeal11** [reply]
Ah I see. In the PDF, it says: 
```
This drawing shows the concept of demote and
continue. The actual connections made are not
displayed correctly.
``` 
Since it didn't have the proper connections it contradicted the actual model of the sim which lead me down the other way of thinking.
